### PR TITLE
DP-21554 DP-21555 Mayflower Map Leaflet Integration

### PR DESCRIPTION
**Pull request by vordude**
**Description:**
This PR integrates the Mayflower work adding leaflet maps to the site.

"Multi Pin" maps as seen on org pages
"Small "static"" maps as seen on location pages.

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-21554
https://jira.mass.gov/browse/DP-21555

**To Test:**
- [ ] Confirm Feature Parity, and to an extent design parity for maps on the stie
- [ ] Org Pages (https://massgovfeature2.prod.acquia-sites.com/orgs/massachusetts-registry-of-motor-vehicles)
- [ ] Location Pages  (https://massgovfeature2.prod.acquia-sites.com/locations/wells-state-park)
- [ ] Location Listing pages (Should still be google maps with no changes) (https://massgovfeature2.prod.acquia-sites.com/orgs/massachusetts-registry-of-motor-vehicles/locations?_page=1)

### CODE REVIEW

**Comment by clairesunstudio** [comment]
46f4949 fixes --> https://github.com/massgov/mayflower/pull/1388#issuecomment-840862455
deploy to feature2 --> https://circleci.com/gh/massgov/openmass

**Comment by clairesunstudio** [comment]
This is fixed in Mayflower but not in feature 2
- [ ] map on location pages hard to pane on mobile

To replicate:
- In a mobile browser (tested in iPhone 12)
- Go to page
Mayflower: https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-20949-google-map-swap-poc/patterns/05-pages-location-general-content/05-pages-location-general-content.html
Feature 2: https://massgovfeature2.prod.acquia-sites.com/locations/auburn-aaa-limited-rmv-services
- Tap inside the map and start moving the map
- The page scrolls on feature 2 while the map is being paned.

**Comment by clairesunstudio** [comment]
Latest deploy --> https://app.circleci.com/pipelines/github/massgov/openmass/8278/workflows/1bf6712a-4e9d-4f0a-93e3-0164da694102
- [ ]  map on location pages hard to pane on mobile
Confirm fixed with latest deploy https://massgovfeature2.prod.acquia-sites.com/locations/auburn-aaa-limited-rmv-services

File: docroot/themes/custom/mass_theme/templates/content/node--contact-information--map.html.twig [hunk 1]
```diff
@@ -71,30 +71,38 @@
  *
  */
 #}
-{% include "@molecules/google-map.twig" with {
-  "googleMap": {
-    "static": true,
+{% include "@molecules/map-leaflet.twig" with {
+  "leafletMap": {
+    "isStatic": false,
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/content/node--contact-information--map.html.twig]:
```suggestion
    "isStatic": true,
```
The scroll bug has not been fixed on mobile. Turning off interactions for the initial release https://jira.mass.gov/browse/DP-22029

File: docroot/modules/custom/mass_map/src/MapLocationFetcher.php [hunk 2]
```diff
@@ -249,21 +249,23 @@ public function getLocations(array $nids) {
...
-          $hours = $hours_paragraph->field_hours_structured->view('default');
-          if ($hours) {
-            $hours = \Drupal::service('renderer')->render($hours);
-            $hours = str_replace('<p>', '', $hours);
-            $hours = str_replace('</p>', '|', $hours);
-            $hours = array_map('trim', explode('|', $hours));
-            foreach ($hours as $hour) {
-              if (!$hour) {
-                continue;
+          if (!is_null($hours_paragraph) &&   !is_null($hours_paragraph->field_hours_structured->entity)) {
```
Comment (tom-fleming) [inline docroot/modules/custom/mass_map/src/MapLocationFetcher.php:252]: Didn't notice during my review @vordude but if we still need this check, we should eliminate that extra space after `&&`

**Pull request merged**